watertap/unit_models/boron_removal.py

```diff
@@ -0,0 +1,1202 @@
+        """
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+        # Set solver options
+        opt = get_solver(solver, optarg)
+
+        # ---------------------------------------------------------------------
+        # Initialize holdup block
+        flags = blk.control_volume.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args,
+        )
+        init_log.info_high("Initialization Step 1 Complete.")
+
+        # Apply guess for unit model vars
+        for t in blk.flowsheet().config.time:
+            # Naive guess (pH = 7)
+            blk.conc_mol_H[t].set_value(1e-4)
+            blk.conc_mol_OH[t].set_value(10**-14 * 1000 / blk.conc_mol_H[t].value)
+            TB = value(
+                blk.control_volume.properties_in[t].conc_mol_phase_comp[
+                    "Liq", blk.boron_name_id
+                ]
+            ) + value(
+                blk.control_volume.properties_in[t].conc_mol_phase_comp[
+                    "Liq", blk.borate_name_id
+                ]
+            )
+            Ratio = 10**-9.21 * 1000 / blk.conc_mol_H[t].value
+            blk.conc_mol_Boron[t].set_value(TB / (1 + Ratio))
+            blk.conc_mol_Borate[t].set_value(TB * Ratio / (1 + Ratio))
+        # ---------------------------------------------------------------------
+
+        # ---------------------------------------------------------------------
+        # Solve unit
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+        init_log.info_high("Initialization Step 2 {}.".format(idaeslog.condition(res)))
+
+        # ---------------------------------------------------------------------
+        # Release Inlet state
+        blk.control_volume.release_state(flags, outlvl + 1)
+        init_log.info("Initialization Complete: {}".format(idaeslog.condition(res)))
+
+        # Rescale internal variables
+        for t in blk.flowsheet().config.time:
+            iscale.set_scaling_factor(
+                blk.conc_mol_OH[t], 100 / blk.conc_mol_OH[t].value
+            )
+            iscale.set_scaling_factor(blk.conc_mol_H[t], 100 / blk.conc_mol_H[t].value)
+            iscale.set_scaling_factor(
+                blk.conc_mol_Boron[t], 100 / blk.conc_mol_Boron[t].value
+            )
+            iscale.set_scaling_factor(
+                blk.conc_mol_Borate[t], 100 / blk.conc_mol_Borate[t].value
+            )
+
+    def outlet_pH(self, time=0):
+        return -log10(value(self.conc_mol_H[time]) / 1000)
+
+    def outlet_pOH(self, time=0):
+        return -log10(value(self.conc_mol_OH[time]) / 1000)
+
+    def propogate_initial_state(self):
+        units_meta = self.config.property_package.get_metadata().get_derived_units
+        i = 0
+        for t in self.control_volume.properties_in:

```

Comment:
Since it seems like we need `i`, it would be better to use `enumerate` here, e.g.:
```suggestion
        for i, t in enumerate(self.control_volume.properties_in):
```

That said, since it seems like you just use this to grab the first time index, you might instead just put this before the loop:
`t0 = self.flowsheet().time.first()`